### DISCUSSION TITLE
Configuration of Second Lagrange Order

### DISCUSSION THREAD

**Discussion by jinca** [idaholab/moose#20758] [Q&A Modules: General] [answered]
Hello dear MOOSE group

Could you please provide some link with the Second Lagrange configuration?
I have run a simulation with the first Lagrange order using only 8 cores. It converged!
When I tried with the second lagrange order, I spent 16 cores instead, and it did not converged.
Maybe it is needed an additional configuration?

Thanks in advance for your help!
Julita

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you try with a single core and with 8 cores? Adding more cores does change how the problem is preconditioned. 
What kind of simulation is it?
Could you please attach the convergence history log for the first order and second order cases.

Thank you
Guillaume

**Reply by jinca** [reply]
Thank you dear Guillaume for the nice and quick answer. 

It is about a transient simulation that calculates the temperature as well as the displacement in 3D.

For the FIRST LAGRANGE order, when I tried using one core in a node with 192GB of memory, the simulation was killed. 
It converged successfully by using 2 cores across 2 nodes,  4 cores across 4 nodes, and 8 cores across 8 nodes. 
I am attaching the output file of the test: 4 cores across 4 nodes. I used the same input file for these tests. The first order is by default.

For the SECOND LAGRANGE order, I tried with 16 cores across 16 nodes, 32 cores across 32 nodes, and 64 cores across 64 nodes.
All of them were killed. I am attaching the slurm file that shows the error by setting the second order which indicates that it is not supported. My setting for the second order in the input file as follows:

[GlobalParams]
  order = SECOND
  family = LAGRANGE
  displacements = 'disp_x disp_y disp_z'
[]

[Variables]
  [temperature]
    order = SECOND
    family = LAGRANGE
    [InitialCondition]
      type = ConstantIC
      value = 293
    []
  []
[]

[bench_4cores_4nodes.txt](https://github.com/idaholab/moose/files/8459123/bench_4cores_4nodes.txt)
[slurm-58665149.txt](https://github.com/idaholab/moose/files/8459124/slurm-58665149.txt)

**Reply by GiudGiud** [reply]
Hello

you need to pass ‘second_order = true’ in the Mesh block

Guillaume

**Reply by GiudGiud** [reply]
Opened an issue #20759
We ll fix it in the near future, in the mean time please see my post above

**Reply by jinca** [reply]
Hi dear Guillaume
It run with this output:

Nonlinear System:
  AD size required:        40
  Num DOFs:                14379581
  Num Local DOFs:          423001
  Variables:               "temperature" { "disp_x" "disp_y" "disp_z" } 
  Finite Element Types:    "LAGRANGE" "LAGRANGE" 
  Approximation Orders:    "FIRST" "SECOND" 

I will try to set SECOND as well in the variable part :)

**Reply by jinca** [reply]
Thank you so much GuidGuid :)
I run successfully my simulation in SECOND ORDER after setting this configuration:
[Mesh]
  second_order = true
  [FileMesh]
    type = FileMeshGenerator
    file = 'mesh.e'
  []
  [Scale]
    type = TransformGenerator
    input = FileMesh
    transform = SCALE
    vector_value = '0.01 0.01 0.01'
  []
...
[Variables]
  [temperature]
    order = SECOND
    [InitialCondition]
      type = ConstantIC
      value = 293
    []
  []
[]
...

You might close this ticket. 
Really appreciate your kind help.
Julita Inca
